firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/SessionReportingCoordinator.java

```diff
@@ -36,19 +36,21 @@
  */
 public class SessionReportingCoordinator implements CrashlyticsLifecycleEvents {
 
+  public interface SendCheck {
+    boolean shouldSendViaDataTransport(int variant);
```

Comment:
Observation: This makes visible that the logic is determined by the variant passed into the interface. I think that it isn't necessary to make that visible because the SessionReportingCooridnator shouldn't be responsible for passing settings around.

There's an opportunity to hide that extra complexity. One way you could do it is by renaming this as `SendStrategy`, and make its interface not take any parameters. Then when constructing this, pass in the value. (If the settings were stateful and ever needed to be updated, you would pass in a new strategy alongside the settings.)

Reply:
Done.